### DISCUSSION TITLE
RCP and CSL

### DISCUSSION THREAD

**Discussion by CodingRays** [openthread/openthread#9960] [Q&A] [answered]
I found that a RCP build will still build the csl functions (for example `SubMac::UpdateCsl` when the CSL receiver is enabled. However, as far as i can tell they are never used anywhere. Is that intentional or could they be removed from a rcp build?

**Comment by Irving-cl** [comment, marked as answer]
Hi @CodingRays,

Note that the CSL_RECEIVER feature currently is not supported under host+rcp mode. It can only run on single chip mode (like ot-cli-xxx, ot-ncp-xxx). The reasons are:
* Functionally, if a device is under host+rcp mode, then the device shouldn't be a low power device since it has a powerful host. It tend to be Thread router and won't behave as an SSED
* Technically, in host+rcp mode, the submac control (when to sleep, when to receive) is still done on the host, so there will be some delay for the RCP to do the actual radio action when the host change the state, which is challenging to CSL.

So the problem here is that `CSL_RECEIVER` option shouldn't be turned on for a RCP build.
